roles/os_net_setup/tasks/main.yml

```diff
@@ -1,53 +1,90 @@
-      kubernetes.core.k8s_info:
-        kubeconfig: "{{ cifmw_openshift_kubeconfig }}"
-        context: "{{ cifmw_openshift_context }}"
-        kind: Secret
-        namespace: openstack
-        name: openstack-config-secret
-      register: osp_secret
 
-- name: Construct auth param for openstack
+- name: Create network resources
+  environment:
+    KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+    PATH: "{{ cifmw_path }}"
+  vars:
+    _template_file: network_command.j2
   block:
-    - name: Set openstack auth fact
-      ansible.builtin.set_fact: # noqa: yaml[line-length]
-        openstack_auth: |
-          {{ ((clouds_yaml.resources[0].data['clouds.yaml']| from_yaml)['clouds']['default'] |
-          combine ({'auth': { 'password' : (osp_secret.resources[0].data['secure.yaml']|b64decode|from_yaml)['clouds']['default']['auth']['password']} }, recursive=true))['auth'] }}
-        region_name: "{{ (clouds_yaml.resources[0].data['clouds.yaml']| from_yaml)['clouds']['default']['region_name'] }}"
-  rescue:
-    - name: Fail if we do not have auth information
-      ansible.builtin.fail:
-        msg: |
-          Failed to Retrive auth Infomation
-          clouds.yaml: {{ clouds_yaml }}
-          osp_secret: {{ osp_secret }}
+    - name: Delete existing networks
+      ansible.builtin.shell: |
+        set -euxo pipefail
+        if [
+            $(oc exec -n openstack openstackclient --
+              openstack network list -c Name -f value |
+              grep -c {{ item.name }}) != 0
+        ];then
+          oc exec -n openstack openstackclient --
+          openstack network delete {{ item.name }}
+        fi
+      loop: "{{ cifmw_os_net_setup_config }}"
+      when: not cifmw_os_net_setup_dry_run | bool
+    - name: Print network creation commands
+      ansible.builtin.debug:
+        msg: "{{ lookup('ansible.builtin.template', _template_file) }}"
+    - name: Create networks
+      ansible.builtin.shell: >
+        {{ lookup('ansible.builtin.template', _template_file) }}
+      when: not cifmw_os_net_setup_dry_run | bool
 
-- name: Get network information
-  openstack.cloud.networks_info:
-    auth: "{{ openstack_auth }}"
-    validate_certs: "{{ cifmw_os_net_setup_verify_tls }}"
-    ca_cert: /etc/pki/tls/certs/ca-bundle.crt
-  register: cifmw_os_net_setup_network_list_out
-  until: cifmw_os_net_setup_network_list_out is not failed
-  retries: "{{ cifmw_os_net_setup_osp_calls_retries }}"
-  delay: "{{ cifmw_os_net_setup_osp_calls_delay }}"
+- name: Create subnet resources
+  environment:
+    KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+    PATH: "{{ cifmw_path }}"
+  vars:
+    _template_file: 'subnet_command.j2'
+  block:
+    - name: Delete existing subnets
+      ansible.builtin.shell: |
+        set -euxo pipefail
+        if [
+            $(oc exec -n openstack openstackclient --
+              openstack subnet list -c Name -f value |
+              grep -c {{ item.1.name }}) != 0
```

Comment:
shouldn't you maybe filter only subnets which belongs to the specific network?

Reply:
done!